server/auth/read_only_capabilities_switcher.test.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ *   Copyright OpenSearch Contributors
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import { Logger, Capabilities } from '../../../../src/core/server';
+import { ReadOnlyCapabilitiesSwitcher } from './read_only_capabilities_switcher';
+
+const logger: Logger = {
+  trace: jest.fn(),
+  debug: jest.fn(),
+  info: jest.fn(),
+  warn: jest.fn(),
+  error: jest.fn(),
+  fatal: jest.fn(),
+  log: jest.fn(),
+  get: jest.fn(),
+};
+
+const capabilites: Capabilities = {
+  navLinks: {},
+  management: {},
+  catalogue: {},
+};
+
+describe('test read-only role evaluation', () => {
+  const authinfo = {
+    roles: ['roleA', 'roleB'],
+  };
+
+  const authinfoDefaultReadonlyRoles = {
+    roles: ['roleA', 'kibana_read_only'],
```

Comment:
If we extract that list of default roles to a more global context as suggested by my other comment, please modify this test file to reference those as opposed to hardcoding them here